### ISSUE TITLE
Docs about github pages

### ISSUE THREAD

**Issue by FernandoBasso** [jekyll/jekyll#3385]
I spent sometime reading http://jekyllrb.com/docs/github-pages/ today. My goal was to be able to test locally. The problem was that even though I was using `jekyll serve --baseurl ''`, it worked for the first time and loaded the page, and then, every time I ran `jekyll build`, stylesheets would not be loaded because the href stylesheets pointed to a url containing the `/project-name` path part.

I think it would be nice to mention in http://jekyllrb.com/docs/github-pages/#project-page-url-structure that when one runs `jekyll serve --baseurl ''`, it will keep watching for changes and rebuilding *automatically* with the *correct* URL, and one *doesn't* need to keep building the sources manually (or use `--watch`), which then prevents links for working correctly when previewing the site locally.

EDIT: I was pointed two links in #jekyll. One of them,

http://jekyllrb.com/docs/configuration/#serve-command-options

clearly explains: "the `serve` sub-command can accept any of the options for the build sub-command, which are then applied to the **site build** which occurs **right before your site is served**." So, I guess that the docs about github pages could just reinforce the idea that running `jekyll serve` builds the site and keeps watching for changes automatically and then point to the above link about the `serve` command, or, at least just point to that link. Not everyone reads the *entire* docs before using jekyll. I guess most people read as they need and sometimes when you are doing something, you still lack knowledge about other related and important concepts it it is not always clear that you are just missing some specific topic in the help pages.

The other link that I was pointed to was

https://byparker.com/blog/2014/clearing-up-confusion-around-baseurl/

Perhaps that one could also be useful, idk.

**Comment by chrisfinazzo** [comment]
I was wondering if someone was going to link @parkr's article about baseurl. It did help to clear up some of the confusion there. To your point, another place I would look for clarification is #2954, as it was around this time that the functionality of `watch` was built in to the `serve` command. From this, we got `--no-watch` as an alternative flag that could be set as needed.

(I should know, I worked on this 😄)

Also, don't worry about not reading everything, I made the same mistake in #3339 (regarding the behavior of `pgrep` and the `--detach` option).

RTM and if you're still not sure, just ask 😄.
